device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -129,17 +130,15 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecuto
                 {
                     this.log.trace("MQTT connection will use sas token based auth");
                     this.iotHubUserPassword = this.config.getSasTokenAuthentication().getRenewedSasToken(false, false);
+                    this.webSocketQueryString = NO_CLIENT_CERT_QUERY_STRING;
                 }
                 else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509_CERTIFICATE)
                 {
-                    if (this.config.isUseWebsocket())
-                    {
-                        //Codes_SRS_MQTTIOTHUBCONNECTION_34_027: [If this function is called while using websockets and x509 authentication, an UnsupportedOperationException shall be thrown.]
-                        throw new UnsupportedOperationException("X509 authentication is not supported over MQTT_WS");
-                    }
-
                     this.log.trace("MQTT connection will use X509 certificate based auth");
 
+                    this.webSocketQueryString = "";
+
+                    //X509 auth over websocket should not include the iothub-no-client-cert flag
```

Comment:
Did you want to move this before line 139?

Reply:
Whoops, yes I do